src/kbc_modules/cc_kbc/mod.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::kbc_modules::{KbcCheckInfo, KbcInterface};
+
+mod attester;
+mod crypto;
+mod kbs_protocol;
+
+use anyhow::*;
+use attester::{detect_tee_type, Attester, TEE};
+use crypto::{hash_chunks, TeeKey};
+use kbs_protocol::message::*;
+
+pub struct Kbc {
+    tee: String,
+    token: Option<String>,
+    nonce: String,
+    tee_key: Option<TeeKey>,
+    attester: Option<Box<dyn Attester + Send + Sync>>,
+}
+
+impl KbcInterface for Kbc {
+    fn check(&self) -> Result<KbcCheckInfo> {
+        Err(anyhow!("Check API of this KBC is unimplemented."))
+    }
+
+    fn decrypt_payload(&mut self, _annotation: &str) -> Result<Vec<u8>> {
+        Err(anyhow!("Decrypt Payload API of this KBC is unimplemented."))
+    }
+}
+
+impl Kbc {
+    pub fn new(_kbs_uri: String) -> Kbc {
+        // Detect TEE type of the current platform.
+        let tee_type = detect_tee_type().ok().unwrap_or(TEE::Unknown);
```

Comment:
Why do we call `ok()` here? I think you can call `unwrap_or()` directly on the `Result<TEE>`.

(or, modify `detect_tee_type` above to return `TEE::Unknown` instead of Err(...). And then there's no need for result/option. )